packages/purpose-process/src/services/readModelService.ts

```diff
@@ -158,9 +158,38 @@ async function buildGetPurposesAggregation(
     : {};
 
   const consumersIdsFilter: ReadModelFilter<Purpose> =
-    ReadModelRepository.arrayToFilter(consumersIds, {
-      "data.consumerId": { $in: consumersIds },
-    });
+    consumersIds.length > 0
+      ? {
+          $or: [
+            { "data.consumerId": { $in: consumersIds } },
+            {
+              $and: [
+                {
+                  $or: [
+                    {
+                      "matchingDelegations.data.delegateId": {
+                        $in: consumersIds,
+                      },
+                    },
+                    {
+                      "matchingDelegations.data.delegatorId": {
+                        $in: consumersIds,
+                      },
+                    },
```

Comment:
We should only check that the delegate is present in the consumerIds, while the delegator shall be equal to the purpose `consumerId`, see the corresponding agreements filter: https://github.com/pagopa/interop-be-monorepo/blob/2705559a0c74b32fbe79b166f7a7414ef22aee04/packages/agreement-process/src/services/readModelService.ts#L372-L373